docs/source/architecture.rst

```diff
@@ -1,17 +1,142 @@
+is not exceeded, the network will work properly. In addition, blocks committed
+by nodes are final, so there are no forks in the network.
+
+All nodes on the network participate in the consensus process by verifying
+candidate blocks and exchanging consensus messages to communicate their progress
+and report on the health of the network. A `primary node` leads the network for
+a period of time, and `secondary nodes` monitor the health of the primary.
+The network switches to a new primary (called a `view change`) at regular
+intervals, as well as when the current primary is faulty.
+
+Sawtooth PBFT runs on each node in the network as a `consensus engine`, a
+separate process that handles consensus-related functionality and communicates
+with the validator through the consensus API.
+
+The following sections describe Sawtooth PBFT architecture:
+
+* :ref:`Network overview <network-overview-label>`: Describes PBFT fault
+  tolerance, view changes, sequence numbers, and the information stored
+  by each node
+
+* :ref:`Consensus messages <consensus-messages-label>`: Explains consensus
+  message structures and message types
+
+* :ref:`Sawtooth PBFT operation <pbft-operation-label>`: Shows how the
+  algorithm handles initialization, normal mode (block processing), and view
+  changes
+
+.. _network-overview-label:
+
+Network Overview
+================
+
+Fault Tolerance
+---------------
+
+A PBFT network consists of nodes that are ordered from 0 to `n-1`, where
+`n` is the total number of nodes in the network.
+
+The PBFT algorithm guarantees network `safety
+<https://en.wikipedia.org/wiki/Liveness#Liveness_and_safety>`__
+as long as the number of faulty nodes remains below the required percentage.
+The maximum number of faulty nodes that the network can tolerate is determined
+by the formula :math:`f = \frac{n - 1}{3}`. In other words, no more than one
+third of the nodes (rounded down) can be out of order or dishonest at any given
+time.
+
+For example, a four-node network (the minimum required number of nodes)
+can tolerate one faulty node. Increasing the size of the network reduces the
+likelihood that that all :math:`\frac{n - 1}{3}` nodes would be faulty at the
+same time.
+
+Network `liveness
+<https://en.wikipedia.org/wiki/Liveness#Liveness_and_safety>`__
+is guaranteed as long as there are less than :math:`2f+1` unreachable nodes,
+counting both node and network failures. That is, at least third of the nodes
+(rounded up) must be reachable at all times.
+
+View Changes: Choosing a New Primary
+------------------------------------
+
+A PBFT network has one leader, called the `primary node`. All other nodes on the
+network are called `secondary nodes`.
+
+* The primary requests new blocks from the validator, notifies the network when
+  a new block is available, and reports consensus results (commit or fail) back
+  to the validator.
+
+* The secondary nodes monitor the health of the primary and request a change
```

Comment:
The most important role of the secondaries is to vote on blocks, which I think is worth mentioning here. Also, "monitor the health of the primary" might be a bit misreading, since that may suggest they something more than what they actually do. All they're doing is checking if the primary is malicious (sent a bad message) or faulty (doesn't do it's job in time), or if the network somehow gets stuck (doesn't make progress for a while). It's really kind of "dumb" in that it's not doing particularly intelligent "monitoring".

Reply:
I'll change this to:

"The secondary nodes vote on the new blocks and send messages if the primary   
  appears unreachable or malicious."